core/templates/cowdata.h

```diff
@@ -133,14 +133,14 @@ class CowData {
 	}
 
 	_FORCE_INLINE_ USize _get_alloc_size(USize p_elements) const {
-		return next_po2(p_elements * sizeof(T));
+		if constexpr ((sizeof(T) & (sizeof(T) - 1)) == 0) { // Is T power of 2.
+			return next_po2(p_elements * sizeof(T) + DATA_OFFSET);
+		} else {
+			return next_po2(p_elements) * sizeof(T) + DATA_OFFSET;
+		}
 	}
 
 	_FORCE_INLINE_ bool _get_alloc_size_checked(USize p_elements, USize *out) const {
-		if (unlikely(p_elements == 0)) {
-			*out = 0;
-			return true;
-		}
```

Comment:
You'll need to restore this if https://github.com/godotengine/godot/pull/91463 is merged

Reply:
I don't think anyone can allocate 9223372036854775807 bytes of memory to overflow x.

Reply:
As long as it's guaranteed it won't be called with 0, as you argued to not restore this above based on that `next_po2` handles it

Reply:
I didn't understand how `next_po2` works then. Now I can see that this would only be a problem when the COW is 32-bit. When using 64 bits, this is not a problem because, as I said, you need a huge amount of memory to overflow.